pages/science/reactors/cvodes.rst

```diff
@@ -0,0 +1,242 @@
+
+The arguments taken by the `CVode()` method is shown below:
+
+.. code-block::
+
+    int CVode(void *cvode_mem, realtype tout, N_Vector yout, realtype *tret, int itask);
+
+There are some interesting things to note about this call to `CVode()`:
+
+- `m_cvode_mem` is a pointer to the block of memory that was allocated and configured during initialization.
+- After execution, `m_y` will contain the computed solution vector, and will later be used to update the `ReactorNet` 
+  to its time-integrated state.
+- The `CV_NORMAL` option tells the solver that it should continue taking internal timesteps until it has reached 
+  user-specified `tout` (or just passed it, in which case solutions are reached by interpolation). This provides the 
+  appropriate functionality for `ReactorNet::advance()`. The alternate option, `CV_ONE_STEP`, tells the solver to take 
+  a single internal step, and is used in `ReactorNet::step()`.
+
+The result of the `CVode()` method is assigned to the `flag` object. `CVode()` returns 1 or 0, correpsonding to 
+a successful or unsuccessful integration, respectively. 
+
+.. code-block::
+
+    int flag = CVode(m_cvode_mem, tout, m_y, &m_time, CV_NORMAL);
+
+Step 6: `FuncEval` class describes ODEs to solve
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+How does `CVODES` know what ODE system it should be solving? 
+
+The ODE system was actually already specified using `CVodeInit()`, one of the methods automatically invoked during the
+`ReactorNet::initialize()` routine. `CVODES` requires that its user provide a C function that defines their ODE, 
+able to compute the right-hand side of the ODE system (dy/dt) for a given value of the independent variable, `t`, 
+and the state vector, `y`. For more information about ODE right-hand side function requirements, 
+see `CVODES User Guide <https://sundials.readthedocs.io/en/latest/cvodes/Usage/SIM.html#user-supplied-functions>`__.
+
+The `CVodesIntegrator` wrapper class provides a useful C++ interface for configuring this C function by pairing with 
+`FuncEval`, an abstract base class for ODE right-hand-side function evaluators (`Documentation 
+<https://cantera.org/documentation/docs-2.4/doxygen/html/d1/dd1/classCantera_1_1FuncEval.html>`__). Classes derived 
+from `FuncEval` will implement the evaluation of the provided ODE system.
+
+An ODE right-hand-side evaluator is always needed in the ODE solution process (it's the only way to describe the system!), and for that reason a `FuncEval` object is a required parameter 
+when initializing any type of `Integrator`.
+
+Let's take a look at how `ReactorNet` implements this `FuncEval` object. `ReactorNet` actually points to itself when 
+defining a `FuncEval` type, meaning it defines *itself* as a `FuncEval` derivative.
+
+Then, `ReactorNet` initializes the `Integrator`, using a reference to itself (as a `FuncEval`) from the 
+`this <https://www.geeksforgeeks.org/this-pointer-in-c/>`__ pointer.
```

Comment:
Can we find a better reference than `geeksforgeeks`? I find their site nearly unusable with all the ads, popups to subscribe, etc.